### PR TITLE
[CI] Add pytest-azurepipelines in ​pandas-dev

### LINKED ISSUE #26601: CI: Use azure-pipelines reporting for pytest data
Azure Pipelines can show errors and warnings clearer, by adding some specific syntax to the log.

We are already using this for the linting, where we output flake8 errors like `##vso[task.logissue type=error;sourcepath=%(path)s;linenumber=%(row)s;columnnumber=%(col)s;code=%(code)s;]%(text)s` (see `ci/code_checks.sh`).

This makes it easier to find the errors in the logs, as they appear in the summary. See for example the `Checks_and_docs` build in https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=12078

We should be using the same for tests, and also "upload" to Azure pipelines the summary of the tests.

This is implemented in this pytest plugin: https://github.com/tonybaloney/pytest-azurepipelines

I guess the best option is to install the plugin (may be install in the azure yaml files, as the dependency is not useful outside of azure). The plugin is quite simple, if it makes sense we may prefer to implement something specific to our needs based on the plugin code: https://github.com/tonybaloney/pytest-azurepipelines/blob/master/pytest_azurepipelines.py

### PR DESCRIPTION
**Pull request by makbigc**
- closes #26601 

Is it better to put `pip install pytest-azurepipelines` inside `setup_env.sh`?

### CODE REVIEW

**Comment by jreback** [comment]
you can try adding this to setup_env.sh.

I think this doesn't work this way because the commandas are executed in different shells (so another option is to do it in the same context

```
activate pandas-dev && pip instal .....
```

**Comment by makbigc** [comment]
@jreback Do you have anything else to implement?

File: ci/deps/azure-35-compat.yaml [hunk 1]
```diff
@@ -28,3 +28,4 @@ dependencies:
     # for python 3.5, pytest>=4.0.2 is not available in conda
     - pytest==4.5.0
     - html5lib==1.0b2
+    - pytest-azurepipelines
```
Comment (TomAugspurger) [inline ci/deps/azure-35-compat.yaml]: Can this be installed from conda-forge?

**Comment by makbigc** [comment]
@TomAugspurger Implemented. Anything else?

**Comment by TomAugspurger** [comment]
The pytest-azurepiplines docs state that it

> Overloads the –junit-xml flag on execution with a default value

Will this break our coverage output for codecov?

**Comment by makbigc** [comment]
I guess no break is done. Codecov report was also generated in this PR. Is it sufficient? Would you tell me any other way to prove?

**Comment by jreback** [comment]
thanks @makbigc 


cc @pandas-dev/pandas-core  we now have a warnings tab in summary: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=13516&view=results

**Pull request merged**